### PR TITLE
Separate north bin and lib. Refactor module layout

### LINKED ISSUE #91: Split runtime into lib and bin part
Split into lib and bin. The iterating part here is the handling of the configuration that cannot be statically read somewhere and must be passed from the bin.

### PR DESCRIPTION
**Pull request by flxo**
Fixes #91

### CODE REVIEW

File: north/src/runtime/linux/cgroups.rs [hunk 1]
```diff
@@ -12,7 +12,8 @@
 //   See the License for the specific language governing permissions and
 //   limitations under the License.
 
-use crate::{Event, EventTx, SETTINGS};
+use super::super::super::runtime::{config, Event, EventTx};
```
Comment (marcmo) [inline north/src/runtime/linux/cgroups.rs:15]: maybe nicer:
```
use crate::runtime::{config, Event, EventTx};
```

File: north/src/runtime/linux/mount.rs [hunk 2]
```diff
@@ -120,44 +120,36 @@ pub async fn mount(
...
+    let c_data = data.as_ref().map_or(ptr::null(), |d| d.as_ptr()) as *const c_void;
+    let result = unsafe {
+        libc::mount(
+            c_source.as_ptr(),
+            c_target.as_ptr(),
+            c_fstype.as_ptr(),
+            flags.bits(),
+            c_data,
+        )
+    };
```
Comment (marcmo) [inline north/src/runtime/linux/mount.rs:133]: why not spawn on the thread pool without blocking the runtime?

File: north/src/runtime/state.rs [hunk 3]
```diff
@@ -283,27 +297,27 @@ impl State {
...
+    //         npk::uninstall(app.container())
+    //             .await
+    //             .map_err(Error::UninstallationError)?;
+    //         self.applications.remove(&app.manifest().name);
+    //         Ok(())
+    //     } else {
+    //         warn!("Cannot uninstall running container {}", app);
+    //         Err(Error::ApplicationRunning(vec![app.manifest().name.clone()]))
+    //     }
+    // }
```
Comment (marcmo) [inline north/src/runtime/state.rs:320]: missing in commit message

File: north/src/runtime/linux/cgroups.rs (hunk 1 above)
Comment (flxo) [reply north/src/runtime/linux/cgroups.rs:15]: The idea is to have inner module `use` directive relative and only needed ones absolute based on the crate root.

File: north/src/runtime/linux/mount.rs (hunk 2 above)
Comment (flxo) [reply north/src/runtime/linux/mount.rs:133]: There's a issue with calling `libc::mount` from a `spawn_blocking` future. @alfonsoros88 investigates here. The mounting code will be refactored in #106 so calling the mounts in the non blocking context here is temporary fine.

File: north/src/runtime/state.rs (hunk 3 above)
Comment (flxo) [reply north/src/runtime/state.rs:320]: Yes. I think I commented it because the install/uninstall/update thing is a subject to be totally reworked.

Comment (marcmo) [reply north/src/runtime/state.rs:320]: agreed...not part of the pr though

File: north/src/runtime/linux/cgroups.rs (hunk 1 above)
Comment (marcmo) [reply north/src/runtime/linux/cgroups.rs:15]: ok...looks weird but you are the expert

**Comment by marcmo** [comment]
could you please increase the version and add a changelog entry @flxo ?

**Comment by flxo** [comment]
> could you please increase the version and add a changelog entry @flxo ?

Done and rebased.

**Pull request merged**